src/redux/store.js

```diff
@@ -1,9 +1,14 @@
 import { configureStore } from "@reduxjs/toolkit";
-import tasksReducer from './tasks'
-
+import watchTodos from "./sagas/todoSaga";
+import tasksReducer from "./slices/tasksSlice";
+import createSagaMiddleware from "redux-saga";
+const sagaMiddleware = createSagaMiddleware();
 
 export default configureStore({
-    reducer:{
-        tasks:tasksReducer
-    }
-})
\ No newline at end of file
+  reducer: {
+    tasks: tasksReducer,
+  },
+  middleware: [sagaMiddleware],
+});
+
+sagaMiddleware.run(watchTodos);
```

Comment:
Instead of using the task reducers directly, the store should receive a combined reducer from the ducks index.js file to make the app more scaleable.